WinPort/src/Backend/TTY/TTYInputSequenceParser.cpp

```diff
@@ -309,6 +309,269 @@ void TTYInputSequenceParser::ParseAPC(const char *s, size_t l)
+		    } else {
+				ir.Event.KeyEvent.dwControlKeyState &= ~LEFT_ALT_PRESSED;
+		    }
+			break;
+		case 57449 : ir.Event.KeyEvent.wVirtualKeyCode = VK_MENU;
+		    if (event_type != KITTY_EVT_KEYUP) {
+				ir.Event.KeyEvent.dwControlKeyState |= RIGHT_ALT_PRESSED;
+		    } else {
+				ir.Event.KeyEvent.dwControlKeyState &= ~RIGHT_ALT_PRESSED;
+		    }
+		    ir.Event.KeyEvent.dwControlKeyState |= ENHANCED_KEY;
+			break;
+		case 57441 : ir.Event.KeyEvent.wVirtualKeyCode = VK_SHIFT;
+		    // todo: add LEFT_SHIFT_PRESSED / RIGHT_SHIFT_PRESSED
+		    // see https://github.com/microsoft/terminal/issues/337
+		    if (event_type != KITTY_EVT_KEYUP) {
+				ir.Event.KeyEvent.dwControlKeyState |= SHIFT_PRESSED;
+		    } else {
+				ir.Event.KeyEvent.dwControlKeyState &= ~SHIFT_PRESSED;
+		    }
+			break;
+		case 57447 : ir.Event.KeyEvent.wVirtualKeyCode = VK_SHIFT;
+		    // todo: add LEFT_SHIFT_PRESSED / RIGHT_SHIFT_PRESSED
+		    // see https://github.com/microsoft/terminal/issues/337
+		    if (event_type != KITTY_EVT_KEYUP) {
+				ir.Event.KeyEvent.dwControlKeyState |= SHIFT_PRESSED;
+		    } else {
+				ir.Event.KeyEvent.dwControlKeyState &= ~SHIFT_PRESSED;
+		    }
+			ir.Event.KeyEvent.wVirtualScanCode = RIGHT_SHIFT_VSC;
+			break;
+
+		case 57360 : ir.Event.KeyEvent.wVirtualKeyCode = VK_NUMLOCK; break;
+		case 57358 : ir.Event.KeyEvent.wVirtualKeyCode = VK_CAPITAL; break;
+		
+	}
+	switch (s[i]) {
+		case 'A': ir.Event.KeyEvent.wVirtualKeyCode = VK_UP;
+			ir.Event.KeyEvent.dwControlKeyState |= ENHANCED_KEY; break;
+		case 'B': ir.Event.KeyEvent.wVirtualKeyCode = VK_DOWN;
+			ir.Event.KeyEvent.dwControlKeyState |= ENHANCED_KEY; break;
+		case 'C': ir.Event.KeyEvent.wVirtualKeyCode = VK_RIGHT;
+			ir.Event.KeyEvent.dwControlKeyState |= ENHANCED_KEY; break;
+		case 'D': ir.Event.KeyEvent.wVirtualKeyCode = VK_LEFT;
+			ir.Event.KeyEvent.dwControlKeyState |= ENHANCED_KEY; break;
+		case 'E': ir.Event.KeyEvent.wVirtualKeyCode = VK_NUMPAD5; break;
+		case 'H': ir.Event.KeyEvent.wVirtualKeyCode = VK_HOME; 
+			ir.Event.KeyEvent.dwControlKeyState |= ENHANCED_KEY; break;
+		case 'F': ir.Event.KeyEvent.wVirtualKeyCode = VK_END;
+			ir.Event.KeyEvent.dwControlKeyState |= ENHANCED_KEY; break;
+		case 'P': ir.Event.KeyEvent.wVirtualKeyCode = VK_F1; break;
+		case 'Q': ir.Event.KeyEvent.wVirtualKeyCode = VK_F2; break;
+		case 'R': ir.Event.KeyEvent.wVirtualKeyCode = VK_F3; break;
+		case 'S': ir.Event.KeyEvent.wVirtualKeyCode = VK_F4; break;
+	}
+
+	if (ir.Event.KeyEvent.wVirtualScanCode == 0) {
+		ir.Event.KeyEvent.wVirtualScanCode = 
+			WINPORT(MapVirtualKey)(ir.Event.KeyEvent.wVirtualKeyCode, MAPVK_VK_TO_VSC);
+	}
+
+	ir.Event.KeyEvent.uChar.UnicodeChar = params[0][1] ? params[0][1] : params[0][0];
+	if (
+		(ir.Event.KeyEvent.uChar.UnicodeChar < 32) ||
+		(ir.Event.KeyEvent.uChar.UnicodeChar == 127) ||
+		((ir.Event.KeyEvent.uChar.UnicodeChar >= 57358) && (ir.Event.KeyEvent.uChar.UnicodeChar <= 57454))
```

Comment:
please use WCHAR_IS_VALID instead

Reply:
this is not validity check, but is filtering of some chars that are used by kitty to identify some function keys (including  some unicode private area code points). they are technically valid and will pass WCHAR_IS_VALID check. but I added validity check also